src/components/LibraryItem.tsx

```diff
@@ -245,4 +246,18 @@ export class LibraryItem extends React.Component<LibraryItemProps, LibraryItemSt
             libraryContainer.raiseEvent("itemClicked", this.props.data.contextData);
         }
     }
+
+    onLibraryItemHoveredOn() {
+        let libraryContainer = this.props.libraryContainer;
+        if (this.props.data.childItems.length == 0) {
+            libraryContainer.raiseEvent("itemHoveredOn", this.props.data.contextData);
```

Comment:
Documenting stuff is a good idea, but that won't help us catch errors at compile time.